app/src/main/java/org/schabi/newpipe/fragments/MainFragment.java

```diff
@@ -189,6 +194,38 @@ private void updateTabsIconAndDescription() {
     private void updateTitleForTab(final int tabPosition) {
         setTitle(tabsList.get(tabPosition).getTabName(requireContext()));
     }
+    private void updateTabsPosition() {
+        final ScrollableTabLayout tabLayout = binding.mainTabLayout;
+        final ViewPager viewPager = binding.pager;
+        final RelativeLayout.LayoutParams tabParams = (RelativeLayout.LayoutParams)
+                tabLayout.getLayoutParams();
+        final RelativeLayout.LayoutParams pagerParams = (RelativeLayout.LayoutParams)
+                viewPager.getLayoutParams();
+        if (PreferenceManager.getDefaultSharedPreferences(requireContext())
+                .getBoolean(getString(R.string.main_tabs_position_key), false)) {
+            tabParams.removeRule(RelativeLayout.ALIGN_PARENT_TOP);
+            tabParams.addRule(RelativeLayout.ALIGN_PARENT_BOTTOM);
+            pagerParams.removeRule(RelativeLayout.BELOW);
+            pagerParams.addRule(RelativeLayout.ABOVE, R.id.main_tab_layout);
+            tabLayout.setSelectedTabIndicatorGravity(TabLayout.INDICATOR_GRAVITY_TOP);
+            final TypedValue typedValue = new TypedValue();
+            getContext().getTheme().resolveAttribute(R.attr.colorPrimary, typedValue,
+                    true);
+            getActivity().getWindow().setNavigationBarColor(typedValue.data);
```

Comment:
The navigation bar seems to remain red even after going to e.g. the search fragment (which doesn't have tabs). Anyway, this comment would become outdated if you just made the tab selector the same color of the app's background color.